### PR TITLE
Refactor build process from Python to JavaScript

### LINKED ISSUE #71: CI 測試的方案覆蓋檢查
CI 測試（test/main.js）最好也檢查下測試自身是否列出了所有的會被建進 index.js 的方案。

因為目前的測試，若有方案漏列，該方案會直接漏過檢查（觀察最近的 PR 發現的）。

### LINKED ISSUE #77: `from字頭` 中 `選項` 引數應為列表
現在 `schema` 是列表，傳回結果中的 `推導結果` 屬性亦為列表，只有 `選項` 引數要求指定 `{ [方案名: string]: 選項 }` 字典，不 consistent。

### PR DESCRIPTION
**Pull request by graphemecluster**
- Replace Python build script with JavaScript
  - The `選項` argument of `from字頭` now accepts both objects and arrays. (Allowing specifying objects is purely for backward compatibility.) Results are outputted as both array elements and named properties. Closes #77.
- Check coverage during building and testing (Fixes #71)
- Separate type definitions to the dedicated file such that they are recognised when called in a TypeScript file
- Move generated artifacts to a `dist` directory and make the npm package include only this directory

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -3,7 +3,6 @@ import taggedIs from './rules/tagged-is.js';
 
 export default [
   {
-    ignores: ['index.js'],
```
Comment (syimyuzya) [inline eslint.config.js:6]: Should be `ignores: ['dist/**/*'],`

File: package.json [hunk 2]
```diff
@@ -3,10 +3,19 @@
...
+			"default": "./dist/index.js",
+			"types": "./dist/index.d.ts"
+		}
+	},
+  "files": ["dist"],
   "scripts": {
-    "lint": "eslint .",
-    "lint:fix": "eslint --fix .",
+    "lint": "eslint . --ignore-pattern dist/**",
+    "lint:fix": "eslint --fix . --ignore-pattern dist/**",
```
Comment (syimyuzya) [inline package.json:17]: The path pattern should go to the ESLint config. (See eslint.config.js)

File: package.json [hunk 3]
```diff
@@ -3,10 +3,19 @@
...
   "type": "module",
-  "main": "index.js",
+	"main": "./dist/index.js",
+	"types": "./dist/index.d.ts",
+	"exports": {
+		".": {
+			"default": "./dist/index.js",
+			"types": "./dist/index.d.ts"
+		}
+	},
```
Comment (syimyuzya) [inline package.json:13]: Inconsistent indentation

**Review by syimyuzya — CHANGES_REQUESTED** [review]

**Comment by syimyuzya** [comment]
Also, that "Python" in the VSCode extension recommendations can be removed (and replaced by ESLint).

File: package.json (hunk 2 above)
Comment (syimyuzya) [reply package.json:17]: Makes sense 😃

Comment (syimyuzya) [inline package.json:17]: The pattern should be `dist/**/*`.

**Review by syimyuzya — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (syimyuzya) [reply package.json:13]:
```suggestion
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "default": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
```

**Comment by graphemecluster** [comment]
@syimyuzya 你 commit 啦，我這幾天 AFK 😄

**Comment by syimyuzya** [comment]
> @syimyuzya 你 commit 啦，我這幾天 AFK 😄

好嗒～

**Review by syimyuzya — APPROVED** [review]

**Pull request merged**